src/main/kotlin/solve/settings/visualization/fields/view/VisualizationSettingsLayerCell.kt

```diff
@@ -190,67 +171,63 @@ class VisualizationSettingsLayerCell(
             }
             layerSettingsButton.action {
                 if (!isPopOverShowing) {
-                    val showPosition = calculatePopOverShowPosition(spawnNode, layerType)
-                    showPopOver(popOver, spawnNode, showPosition)
+                    popOver.show()
                 } else {
                     popOver.hide()
                 }
             }
         }
     }
 
-    private fun showPopOver(popOver: PopOver, spawnNode: Node, showPosition: DoublePoint) {
-        popOver.detach()
-        popOver.show(spawnNode, showPosition.x, showPosition.y)
-    }
-
-    private fun createLayerSettingsPopOverNode(layerSettings: LayerSettings): Node? =
+    private fun createLayerSettingsPopOverNode(layerSettings: LayerSettings, title: String): SettingsDialogNode? =
         when (getLayerSettingsType(layerSettings)) {
             LandmarkType.Keypoint ->
                 PointLayerSettingsPopOverNode(
                     layerSettings as LayerSettings.PointLayerSettings,
-                    sceneController
+                    sceneController,
+                    title,
+                    dialogClosingController
                 ).getPopOverNode()
+
             LandmarkType.Line ->
                 LineLayerSettingsPopOverNode(
                     layerSettings as LayerSettings.LineLayerSettings,
-                    sceneController
+                    sceneController,
+                    title,
+                    dialogClosingController
                 ).getPopOverNode()
+
             LandmarkType.Plane -> null
         }
 
-    private fun createLayerSettingsPopOver(contentNode: Node, titleLabel: String): PopOver {
+    private fun createLayerSettingsPopOver(contentNode: Node, titleLabel: String): MFXStageDialog {
+        val content = createGenericDialog(contentNode)
+        val dialog = createStageDialog(content, sceneView.currentStage, sceneView.root)
+
         val popOver = PopOver(contentNode)
         popOver.detach()
         popOver.title = titleLabel
 
         Platform.runLater {
-            // Needed for the safe window closing.
-            listView.scene?.window?.setOnCloseRequest {
-                popOver.hide(Duration.ZERO)
+            dialogClosingController.isClosing.onChange {
+                if (it) dialog.close()
             }
```

Comment:
Don't use controller for that, you can use observable property. Controller is not correct here, because it is singleton and you need to create separate scopes.